### PR TITLE
Add broker skew and cluster list routes

### PR DESCRIPTION
**Pull request by kelseylam**
Added routes to conf/routes to get back the broker skew for a given cluster and topic in a JSON and a list of clusters (represented by dictionaries) in a JSON. Added methods to KafkaStateCheck controller attached to the routes.

### CODE REVIEW

File: app/controllers/api/KafkaStateCheck.scala [hunk 1]
```diff
@@ -41,6 +42,37 @@ class KafkaStateCheck (val messagesApi: MessagesApi, val kafkaManagerContext: Ka
...
+        clusterList => Ok(getClusterListJson(clusterList, status))
+      )
+    }
+  }
+
+  def getClusterListJson(clusterList: KMClusterList, status: Option[String]) = {
+    val active = clusterList.active.map(cc => Map("name" -> cc.name, "status" -> "active"))
+    val pending = clusterList.pending.map(cc => Map("name" -> cc.name, "status" -> "pending"))
+
+    if (status.getOrElse(None) == "active") {
```
Comment (arijitdey) [inline app/controllers/api/KafkaStateCheck.scala]: I didn't understand this part.

**Comment by arijitdey** [comment]
LGTM

File: app/controllers/api/KafkaStateCheck.scala [hunk 2]
```diff
@@ -1,20 +1,21 @@
 /**
- * Copyright 2015 Yahoo Inc. Licensed under the Apache License, Version 2.0
- * See accompanying LICENSE file.
- */
+  * Copyright 2015 Yahoo Inc. Licensed under the Apache License, Version 2.0
```
Comment (realstraw) [inline app/controllers/api/KafkaStateCheck.scala]: why? we should value consistency more, if the rest of the code base is formatted this way, let's keep it this way. Also, most of other Krux's scala code is following javadoc style.

File: app/controllers/api/KafkaStateCheck.scala (hunk 1 above)
Comment (realstraw) [inline app/controllers/api/KafkaStateCheck.scala]: People would expect `status.getOrElse` to return a String, but `status.getOrElse(None)` would return `Any` as `None` is of type `Option. You should either use pattern matching or compare with the right types. e.g.

pattern matching (preferred)

```scala
status match {
  case Some("active") => ...
  case Some("pending") => ...
  case _ => ...
}
```

or `if (status == Some("active"))` also works but not preferred

**Comment by arijitdey** [comment]
👍

File: app/controllers/api/KafkaStateCheck.scala [hunk 3]
```diff
@@ -7,14 +7,15 @@ package controllers.api
...
+import kafka.manager.model.ActorModel.KMClusterList
 import models.navigation.Menus
 import play.api.i18n.{I18nSupport, MessagesApi}
 import play.api.libs.json._
 import play.api.mvc._
 
 /**
- * @author jisookim0513
- */
+  * @author jisookim0513
```
Comment (realstraw) [inline app/controllers/api/KafkaStateCheck.scala]: this one as well (no need to change the doc string)

**Comment by realstraw** [comment]
👍

File: app/controllers/api/KafkaStateCheck.scala [hunk 4]
```diff
@@ -41,6 +42,35 @@ class KafkaStateCheck (val messagesApi: MessagesApi, val kafkaManagerContext: Ka
...
+    val pending = clusterList.pending.map(cc => Map("name" -> cc.name, "status" -> "pending"))
+
+    status match {
+      case Some("active") => Json.obj("clusters" -> active.sortBy(_("name")))
+      case Some("pending") => Json.obj("clusters" -> pending.sortBy(_("name")))
+      case _ => Json.obj("clusters" -> (active ++ pending).sortBy(_("name")))
+    }
+  }
+
+  def brokersSkewPercentage(c: String, t:String) = Action.async { implicit request =>
```
Comment (jnewman) [inline app/controllers/api/KafkaStateCheck.scala]: Nit - space after the colon please

File: app/controllers/api/KafkaStateCheck.scala [hunk 5]
```diff
@@ -41,6 +42,35 @@ class KafkaStateCheck (val messagesApi: MessagesApi, val kafkaManagerContext: Ka
...
+    kafkaManager.getTopicIdentity(c, t).map { errorOrTopicIdentity =>
```
Comment (jnewman) [inline app/controllers/api/KafkaStateCheck.scala]: Nit - could be `_.fold(error => ...`

File: app/controllers/api/KafkaStateCheck.scala [hunk 6]
```diff
@@ -41,6 +42,35 @@ class KafkaStateCheck (val messagesApi: MessagesApi, val kafkaManagerContext: Ka
     }
   }
 
+  def clusters(status: Option[String]) = Action.async { implicit request =>
+    kafkaManager.getClusterList.map { errorOrClusterList =>
+      errorOrClusterList.fold(
```
Comment (jnewman) [inline app/controllers/api/KafkaStateCheck.scala]: Same as below, don't really need to define a callback, could be `_.fold(...`

**Comment by jnewman** [comment]
Just nits :+1:

File: app/controllers/api/KafkaStateCheck.scala (hunk 5 above)
Comment (kelseylam) [inline app/controllers/api/KafkaStateCheck.scala]: Should we still use _.fold if the rest of the existing code in this file seems to explicitly just say errorOrTopicIdentity.fold in this context?

Comment (jnewman) [inline app/controllers/api/KafkaStateCheck.scala]: Stick with prevailing style and ignore me :-)

**Pull request merged**